# TASCA M9 T01

## Objectivos
- Realizar minería de textos. 
- Interpretación de los textos. 
- Análisis de los resultados.


## Duración: 2 días

## Descripción
Ejercicios de análisis de sentimiento y textos.

<font color='blue'>__.........................................................................................................................................................................................................................................................................__

# Ejercicio 1
Coge un texto en inglés que quieras, y calcula la frecuencia de sus palabras.



In [ ]:
# Insatalr python-docx para poder descargar archivos en word
pip install python-docx

In [4]:
from docx import Document
# Cargar el documento Word
documento = Document('/Users/Esteban/OneDrive/DATOS E/Curso 2309 DATA SCIENCE (IT-ACAD Sept 23 a - 24)/Sprint 9.- Análisis del SENTIMIENTO/TASCA M9 T01/Texto OurManyTrespasses1-22rev.docx')
documento

In [6]:
# Extraer el texto del documento

texto = ""# Inicializo una cadena vacía llamada Texto

for paragraph in documento.paragraphs: # Recorro todos los párrafos del documento
    texto += paragraph.text + "\n" # Agrego cada párrafo a la cadena Texto y le añado salto de línea con \n 

# Imprimir todo el texto extraído:    
texto   

"Our Many Trespasses\nEileen Legaspi-Ramirez\n\nMuntadas’s Manila-Sevilla: Exercises on Past and Present Memories happens upon Manila amidst a toxic cacophony of factoid-pandering. There is the shrill pre-election maneuvering of shady allegiances among political parties, the continued militarizing of pandemic health and safety rhetoric with consequent clamping upon oppositional forces on the left and left of center, the playing off of superpower antics on the regional and world stage, all these on top of the hope-wrenching heft of having to survive day in and day out. Fairly nearby, India, Indonesia, Thailand, and Myanmar are also groaning from the staggering toll of Covid-19, understated death and infection figures making absurd any sense of intentional slack. One could easily be tempted to ask what the point would be in reckoning with colonial traces as the sheer immediacy of hunger, death, and paralysis stares us down with merciless tenacity.\n\nAnd yet, residual cores of the cultur

In [7]:
# Importar librerías que voy a necesitar: 

import pandas as pd
import nltk
from nltk.sentiment.vader import SentimentIntensityAnalyzer
from nltk.corpus import stopwords
from nltk.tokenize import word_tokenize
from nltk.stem import WordNetLemmatizer

In [8]:
# Tokenizar el texto en palabras con word_tokenize convirtiendo las mayúsculas en minúsculas.
# Si quisiera mantener las mayúsculas del texto original, ultilizaría el código: palabras = word_tokenize(texto)
palabras = word_tokenize(texto.lower())
palabras

['our',
 'many',
 'trespasses',
 'eileen',
 'legaspi-ramirez',
 'muntadas',
 '’',
 's',
 'manila-sevilla',
 ':',
 'exercises',
 'on',
 'past',
 'and',
 'present',
 'memories',
 'happens',
 'upon',
 'manila',
 'amidst',
 'a',
 'toxic',
 'cacophony',
 'of',
 'factoid-pandering',
 '.',
 'there',
 'is',
 'the',
 'shrill',
 'pre-election',
 'maneuvering',
 'of',
 'shady',
 'allegiances',
 'among',
 'political',
 'parties',
 ',',
 'the',
 'continued',
 'militarizing',
 'of',
 'pandemic',
 'health',
 'and',
 'safety',
 'rhetoric',
 'with',
 'consequent',
 'clamping',
 'upon',
 'oppositional',
 'forces',
 'on',
 'the',
 'left',
 'and',
 'left',
 'of',
 'center',
 ',',
 'the',
 'playing',
 'off',
 'of',
 'superpower',
 'antics',
 'on',
 'the',
 'regional',
 'and',
 'world',
 'stage',
 ',',
 'all',
 'these',
 'on',
 'top',
 'of',
 'the',
 'hope-wrenching',
 'heft',
 'of',
 'having',
 'to',
 'survive',
 'day',
 'in',
 'and',
 'day',
 'out',
 '.',
 'fairly',
 'nearby',
 ',',
 'india',
 ',',
 'indo

In [9]:
# Obtener el número total de palabras.
numero_total_palabras = len(palabras)

# Imprimir el resultado
print(f'El número total de palabras es: {numero_total_palabras}')

El número total de palabras es: 3107


In [10]:
# Calculo la frecuencia de las palabras con FreqDist

from nltk import FreqDist

frecuencia_palabras = FreqDist(palabras)

for palabra, frecuencia in frecuencia_palabras.items():
    print(f'{palabra}: {frecuencia}')

our: 3
many: 4
trespasses: 1
eileen: 1
legaspi-ramirez: 1
muntadas: 17
’: 45
s: 30
manila-sevilla: 8
:: 7
exercises: 4
on: 25
past: 7
and: 94
present: 11
memories: 4
happens: 1
upon: 10
manila: 7
amidst: 6
a: 39
toxic: 2
cacophony: 1
of: 116
factoid-pandering: 1
.: 90
there: 6
is: 22
the: 150
shrill: 1
pre-election: 1
maneuvering: 1
shady: 1
allegiances: 1
among: 6
political: 2
parties: 1
,: 109
continued: 1
militarizing: 1
pandemic: 5
health: 3
safety: 1
rhetoric: 2
with: 20
consequent: 2
clamping: 1
oppositional: 1
forces: 4
left: 2
center: 1
playing: 1
off: 9
superpower: 1
antics: 1
regional: 1
world: 1
stage: 1
all: 6
these: 10
top: 1
hope-wrenching: 1
heft: 1
having: 1
to: 62
survive: 2
day: 2
in: 72
out: 3
fairly: 2
nearby: 1
india: 2
indonesia: 1
thailand: 1
myanmar: 1
are: 9
also: 3
groaning: 1
from: 15
staggering: 1
toll: 1
covid-19: 3
understated: 1
death: 2
infection: 1
figures: 1
making: 3
absurd: 1
any: 5
sense: 4
intentional: 1
slack: 1
one: 8
could: 3
easily: 3
be: 14
te

In [11]:
# Visualización de las frecuencias en dataframe:

# Convertir FreqDist a DataFrame
df_frecuencia = pd.DataFrame(list(frecuencia_palabras.items()), columns=['Palabra', 'Frecuencia'])

# Ordenar DataFrame por frecuencia descendente
df_frecuencia = df_frecuencia.sort_values(by='Frecuencia', ascending=False)

# Mostrar el DataFrame
df_frecuencia

Palabra  Frecuencia
28         the         150
23          of         116
37           ,         109
13         and          94
25           .          90
...        ...         ...
593     places           1
594     guests           1
595     tables           1
596    grafted           1
1369  rewrites           1

[1370 rows x 2 columns]

### OBTENCIÓN DE FRECUENCIAS

El dataframe anterior ofrede las frecuencias de todas las palabras, signos de puntuación y otros elementos que aparecen en el texto

### DESARROLLO DE UN INPUT PARA LOCALIZAR LA FRECUENCIA DE UNA PALABRA

Teniendo en cuenta las frecuencias anteriores, voy a desarrollar un código de un imput que me permita buscar la frecuecia de cualquier palabra o sigino que aparezca en el texto.

In [12]:
# Para buscar la frecuencia de una palabra en concreto:

# Solicitar al usuario que ingrese la palabra
palabra_buscada = input("Ingresa una palabra: ")

# Convertir la palabra ingresada a minúsculas
palabra_buscada = palabra_buscada.lower()

# Buscar la frecuencia en el DataFrame
frecuencia_palabra = df_frecuencia.loc[df_frecuencia['Palabra'].str.lower() == palabra_buscada, 'Frecuencia'].values

if len(frecuencia_palabra) > 0:
    # Imprimir el resultado si se encontró la palabra
    print(f'La frecuencia de la palabra "{palabra_buscada}" es: {frecuencia_palabra[0]}')
else:
    # Imprimir un mensaje si la palabra no se encuentra en el DataFrame
    print(f'La palabra "{palabra_buscada}" no se encontró en el DataFrame.')
    

Ingresa una palabra: Manila
La frecuencia de la palabra "manila" es: 7


<font color='blue'>__.........................................................................................................................................................................................................................................................................__
  

# Ejercicio 2
Saca las stopwords y realiza stemming en tu conjunto de datos.

In [14]:
# Filtrar palabras vacías (stopwords) en dos pasos

# 1º.- Obtengo una lista de palabras vacías (stopwords) en inglés desde la biblioteca NLTK y la guardo en stop_words.
stop_words = set(stopwords.words('english'))  

# 2ª.- Filtro las palabras del texto original de la siguiente forma: 
# 2.1.- word.lower(): Convierte cada palabra a minúsculas para asegurar uniformidad en el análisis y evitar duplicaciones
# 2.2.- word.isalpha(): Verifica si la palabra contiene solo caracteres alfabéticos (letras) y no incluye ==>
# ==> caracteres especiales o números. Esto ayuda a eliminar signos de puntuación y números del análisis.
# word.lower() not in stop_words: Verifica si la palabra no está en la lista de palabras vacías ==>
# Si está en la lista de palabras vacías, se excluye de la lista final.

palabras_filtradas = [word.lower() for word in palabras if word.isalpha() and word.lower() not in stop_words]

palabras_filtradas

['many',
 'trespasses',
 'eileen',
 'muntadas',
 'exercises',
 'past',
 'present',
 'memories',
 'happens',
 'upon',
 'manila',
 'amidst',
 'toxic',
 'cacophony',
 'shrill',
 'maneuvering',
 'shady',
 'allegiances',
 'among',
 'political',
 'parties',
 'continued',
 'militarizing',
 'pandemic',
 'health',
 'safety',
 'rhetoric',
 'consequent',
 'clamping',
 'upon',
 'oppositional',
 'forces',
 'left',
 'left',
 'center',
 'playing',
 'superpower',
 'antics',
 'regional',
 'world',
 'stage',
 'top',
 'heft',
 'survive',
 'day',
 'day',
 'fairly',
 'nearby',
 'india',
 'indonesia',
 'thailand',
 'myanmar',
 'also',
 'groaning',
 'staggering',
 'toll',
 'understated',
 'death',
 'infection',
 'figures',
 'making',
 'absurd',
 'sense',
 'intentional',
 'slack',
 'one',
 'could',
 'easily',
 'tempted',
 'ask',
 'point',
 'would',
 'reckoning',
 'colonial',
 'traces',
 'sheer',
 'immediacy',
 'hunger',
 'death',
 'paralysis',
 'stares',
 'us',
 'merciless',
 'tenacity',
 'yet',
 'residual',


In [15]:
# Obtener el número total de palabras filtradas.
numero_total_palabras_filtradas = len(palabras_filtradas)

# Imprimir el resultado
print(f'El número total de palabras filtradas es: {numero_total_palabras_filtradas}')

El número total de palabras filtradas es: 1630


### NOTA SOBRE LA REDUCCIÓN DE PALABRAS CON EL STOPWORDS: 

Observamos que el total de palabras se ha reducido: 
- de 3107 palabras totales en el texto original.
- a 1630 palabras tras realizar el stopwords.
El número total de palabras se ha reducico prácticamente al 50%.

In [17]:
# Realizar el stemming: eliminar los sufijos de palabras, como "ing" o "ed", para reducirlas a su forma básica.

from nltk.stem import PorterStemmer

# Crear un objeto PorterStemmer para inglés
stemmer = PorterStemmer()


# Aplicar stemming a las palabras filtradas en inglés
palabras_stemmed = [stemmer.stem(palabra) for palabra in palabras_filtradas]

# Imprimir las palabras originales y sus formas stem
for original, stemmed in zip(palabras_filtradas, palabras_stemmed):
    print(f'Original: {original} | Stemmed: {stemmed}')

Original: many | Stemmed: mani
Original: trespasses | Stemmed: trespass
Original: eileen | Stemmed: eileen
Original: muntadas | Stemmed: muntada
Original: exercises | Stemmed: exercis
Original: past | Stemmed: past
Original: present | Stemmed: present
Original: memories | Stemmed: memori
Original: happens | Stemmed: happen
Original: upon | Stemmed: upon
Original: manila | Stemmed: manila
Original: amidst | Stemmed: amidst
Original: toxic | Stemmed: toxic
Original: cacophony | Stemmed: cacophoni
Original: shrill | Stemmed: shrill
Original: maneuvering | Stemmed: maneuv
Original: shady | Stemmed: shadi
Original: allegiances | Stemmed: allegi
Original: among | Stemmed: among
Original: political | Stemmed: polit
Original: parties | Stemmed: parti
Original: continued | Stemmed: continu
Original: militarizing | Stemmed: militar
Original: pandemic | Stemmed: pandem
Original: health | Stemmed: health
Original: safety | Stemmed: safeti
Original: rhetoric | Stemmed: rhetor
Original: consequent |

In [18]:
# Crear un DataFrame con las palabras originales y sus formas stem
df_stemmed = pd.DataFrame({
    'Original': palabras_filtradas,
    'Stemmed': palabras_stemmed
})

# Imprimir el DataFrame
df_stemmed

Original   Stemmed
0           many      mani
1     trespasses  trespass
2         eileen    eileen
3       muntadas   muntada
4      exercises   exercis
...          ...       ...
1625     detours    detour
1626      course     cours
1627     atlases     atlas
1628     begging       beg
1629    rewrites    rewrit

[1630 rows x 2 columns]

In [19]:
# Lemmatize the tokens

from nltk.stem import WordNetLemmatizer
    
lemmatizer = WordNetLemmatizer()
palabras_lematizadas = [lemmatizer.lemmatize(palabra) for palabra in palabras_filtradas]

# Imprimir las palabras originales y sus formas lematizadas
for original, lemmatized in zip(palabras_filtradas, palabras_lematizadas):
    print(f'Original: {original} | Lemmatized: {lemmatized}')

Original: many | Lemmatized: many
Original: trespasses | Lemmatized: trespass
Original: eileen | Lemmatized: eileen
Original: muntadas | Lemmatized: muntadas
Original: exercises | Lemmatized: exercise
Original: past | Lemmatized: past
Original: present | Lemmatized: present
Original: memories | Lemmatized: memory
Original: happens | Lemmatized: happens
Original: upon | Lemmatized: upon
Original: manila | Lemmatized: manila
Original: amidst | Lemmatized: amidst
Original: toxic | Lemmatized: toxic
Original: cacophony | Lemmatized: cacophony
Original: shrill | Lemmatized: shrill
Original: maneuvering | Lemmatized: maneuvering
Original: shady | Lemmatized: shady
Original: allegiances | Lemmatized: allegiance
Original: among | Lemmatized: among
Original: political | Lemmatized: political
Original: parties | Lemmatized: party
Original: continued | Lemmatized: continued
Original: militarizing | Lemmatized: militarizing
Original: pandemic | Lemmatized: pandemic
Original: health | Lemmatized: h

In [20]:
# Crear un DataFrame con las palabras originales y sus formas Lematizadas
df_lemmatized = pd.DataFrame({
    'Original': palabras_filtradas,
    'lemmatized': palabras_lematizadas
})

# Imprimir el DataFrame
df_lemmatized

Original lemmatized
0           many       many
1     trespasses   trespass
2         eileen     eileen
3       muntadas   muntadas
4      exercises   exercise
...          ...        ...
1625     detours     detour
1626      course     course
1627     atlases      atlas
1628     begging    begging
1629    rewrites    rewrite

[1630 rows x 2 columns]

### OBSERVACIONES AL STEAMING Y LEMATIZACIÓN: 

Obtenemos dos tablas, ambas de 1530 registros, que contienen las propuestas de steaming y lematización para cada palabra.

<font color='blue'>__.........................................................................................................................................................................................................................................................................__

# Exercicio 3
Realiza análisis de sentimiento en tu conjunto de datos.

### TEST nº 1: análisis de sentimientos sobre las palabras individualizadas

En un primer paso, realizaré el análisis de sentimientos sobre las palabras individualizadas, tras la Tokenización, eliminación de spotwords, Stemming y Lemmatization.


In [21]:
# Mediante el analizador de sentimientos Vader para analizar el sentimiento del texto (positivo o negativo).
# Hay que hacerlo en el texto preprocesado, tras la Tokenización, eliminación de spotwords, Stemming y Lemmatization

import ssl
from nltk.sentiment import SentimentIntensityAnalyzer
import pandas as pd

# Desactivar la verificación del certificado SSL
ssl._create_default_https_context = ssl._create_unverified_context

# Descargar el lexicon VADER
nltk.download('vader_lexicon')

# Restaurar la verificación del certificado SSL a su estado predeterminado
ssl._create_default_https_context = ssl._create_default_https_context

# Inicializar el analizador de sentimientos
sid = SentimentIntensityAnalyzer()

# Supongamos que tienes una lista llamada 'palabras_filtradas' con tus palabras preprocesadas
# Utiliza las palabras lematizadas para obtener un resultado más representativo
datos = palabras_lematizadas

# Aplicar el análisis de sentimiento a cada palabra y almacenar los resultados en una lista
# resultados_sentimiento = [sid.polarity_scores(palabra) for palabra in datos]

# Aplicar el análisis de sentimiento a cada palabra y almacenar los resultados en una lista de diccionarios
resultados_sentimiento = [{"Palabra": palabra, **sid.polarity_scores(palabra)} for palabra in datos]

# Convertir la lista de resultados a DataFrame
df_sentimiento = pd.DataFrame(resultados_sentimiento)

# Imprimir el DataFrame con los resultados
df_sentimiento


# NOTA:  La columna "compound" se utiliza a menudo como medida general de la polaridad del sentimiento, 
# donde valores mayores a 0 indican sentimientos positivos, menores a 0 negativos y 0 indica neutralidad.

[nltk_data] Downloading package vader_lexicon to
[nltk_data]     /Users/Esteban/nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


Palabra  neg  neu  pos  compound
0         many  0.0  1.0  0.0       0.0
1     trespass  0.0  1.0  0.0       0.0
2       eileen  0.0  1.0  0.0       0.0
3     muntadas  0.0  1.0  0.0       0.0
4     exercise  0.0  1.0  0.0       0.0
...        ...  ...  ...  ...       ...
1625    detour  0.0  1.0  0.0       0.0
1626    course  0.0  1.0  0.0       0.0
1627     atlas  0.0  1.0  0.0       0.0
1628   begging  0.0  1.0  0.0       0.0
1629   rewrite  0.0  1.0  0.0       0.0

[1630 rows x 5 columns]

In [22]:
df_sentimiento.describe()

neg          neu          pos     compound
count  1630.000000  1630.000000  1630.000000  1630.000000
mean      0.029448     0.931902     0.038037     0.003165
std       0.169110     0.251992     0.191344     0.100717
min       0.000000     0.000000     0.000000    -0.680800
25%       0.000000     1.000000     0.000000     0.000000
50%       0.000000     1.000000     0.000000     0.000000
75%       0.000000     1.000000     0.000000     0.000000
max       1.000000     1.000000     1.000000     0.636900

### CONCLUSIÓN AL TEST nº1: 

Vemos que la inmensa gran mayoría de las palabras de consideran de tipo neutro.

Procedo por tanto a calcular los porcentajes que el análisis de sensibilidad arroja para palabras positivas, negativas y neutras:

In [23]:
# Contar el número de 1 en cada columna
num_positivos = (df_sentimiento['pos'] == 1).sum()
num_negativos = (df_sentimiento['neg'] == 1).sum()
num_neutros = (df_sentimiento['neu'] == 1).sum()
num_compounds = (df_sentimiento['compound'] == 1).sum()

# Imprimir los resultados
print(f"Número de positivos: {num_positivos}")
print(f"Número de negativos: {num_negativos}")
print(f"Número de neutros: {num_neutros}")
print(f"Número de compounds: {num_compounds}")

Número de positivos: 62
Número de negativos: 48
Número de neutros: 1519
Número de compounds: 0


In [24]:
# Crear un DataFrame con los recuentos
df_recuentos = pd.DataFrame({
    'Sentimiento': ['Positivos', 'Negativos', 'Neutros', 'Compounds'],
    'Recuento': [num_positivos, num_negativos, num_neutros, num_compounds]
})

# Imprimir el DataFrame con los recuentos
df_recuentos

Sentimiento  Recuento
0   Positivos        62
1   Negativos        48
2     Neutros      1519
3   Compounds         0

In [25]:
# Calcular el total de todas las categorías
total_general = df_recuentos['Recuento'].sum()

# Calcular el porcentaje para cada categoría
df_recuentos['Porcentaje'] = (df_recuentos['Recuento'] / total_general) * 100

# Imprimir el DataFrame con los recuentos y porcentajes
df_recuentos

Sentimiento  Recuento  Porcentaje
0   Positivos        62    3.806016
1   Negativos        48    2.946593
2     Neutros      1519   93.247391
3   Compounds         0    0.000000

In [26]:
# Redondear los porcentajes a un decimal
df_recuentos['Porcentaje'] = df_recuentos['Porcentaje'].round(1)

# Agregar el símbolo '%' a los porcentajes
df_recuentos['Porcentaje'] = df_recuentos['Porcentaje'].astype(str) + '%'

# Imprimir el DataFrame con los recuentos y porcentajes formateados
df_recuentos

Sentimiento  Recuento Porcentaje
0   Positivos        62       3.8%
1   Negativos        48       2.9%
2     Neutros      1519      93.2%
3   Compounds         0       0.0%

### COCLUSIÓN AL ANÁLISIS DE SENTIMIENTO POR PALABRAS.

Según nos muestra la tabla anterior, el análisis de sentimientos por palabras revela que: 
- El 93,2% de las palabras arrojan un sentimiento neutro.
- El 2,9% de las palabras arrojan un sentimiento negativo.
- 3l 3,8% de las palabras arrojan un sentimiento positivo.

<font color='blue'>__.........................................................................................................................................................................................................................................................................__

### TEST nº 2
### REALIZAR EL ANÁLISIS DE SENTIMIENTOS POR FRASES EN EL TEXTO SIN PREPROCESAR

A la vista del resultado anterior, me permito realizar un análisis de sentimiento en las frases del texto, no en las palabras para ver si arroja resultados diferentes al análisis por palabras.

In [27]:
# Hay que dividir el texto en oraciones, en lugar de palabras
import nltk

# Descargar el tokenizer de oraciones
nltk.download('punkt')

# Tokenizar el texto en oraciones
oraciones = nltk.sent_tokenize(texto)

# Imprimir las oraciones
oraciones

[nltk_data] Downloading package punkt to /Users/Esteban/nltk_data...
[nltk_data]   Package punkt is already up-to-date!


['Our Many Trespasses\nEileen Legaspi-Ramirez\n\nMuntadas’s Manila-Sevilla: Exercises on Past and Present Memories happens upon Manila amidst a toxic cacophony of factoid-pandering.',
 'There is the shrill pre-election maneuvering of shady allegiances among political parties, the continued militarizing of pandemic health and safety rhetoric with consequent clamping upon oppositional forces on the left and left of center, the playing off of superpower antics on the regional and world stage, all these on top of the hope-wrenching heft of having to survive day in and day out.',
 'Fairly nearby, India, Indonesia, Thailand, and Myanmar are also groaning from the staggering toll of Covid-19, understated death and infection figures making absurd any sense of intentional slack.',
 'One could easily be tempted to ask what the point would be in reckoning with colonial traces as the sheer immediacy of hunger, death, and paralysis stares us down with merciless tenacity.',
 'And yet, residual cores

In [29]:
import pandas as pd
from nltk.sentiment import SentimentIntensityAnalyzer

# Inicializar el analizador de sentimientos
sid = SentimentIntensityAnalyzer()

# Aplicar el análisis de sentimientos a cada oración y almacenar los resultados en una lista de diccionarios
resultados_sentimiento_oraciones = [{"Oracion": oracion, **sid.polarity_scores(oracion)} for oracion in oraciones]

# Convertir la lista de resultados a DataFrame
df_sentimiento_oraciones = pd.DataFrame(resultados_sentimiento_oraciones)

# Imprimir el DataFrame con los resultados
df_sentimiento_oraciones

Oracion    neg    neu    pos  \
0   Our Many Trespasses\nEileen Legaspi-Ramirez\n\...  0.000  1.000  0.000   
1   There is the shrill pre-election maneuvering o...  0.000  0.866  0.134   
2   Fairly nearby, India, Indonesia, Thailand, and...  0.126  0.874  0.000   
3   One could easily be tempted to ask what the po...  0.158  0.777  0.064   
4   And yet, residual cores of the cultural sphere...  0.000  1.000  0.000   
..                                                ...    ...    ...    ...   
92              Nine dashes past unbuttressed waters.  0.000  1.000  0.000   
93  Tossing and turning through the turbulences of...  0.000  1.000  0.000   
94  All that inevitably entangles dispersed souls ...  0.000  0.822  0.178   
95                              Impasses and detours.  0.000  1.000  0.000   
96           Off course atlases begging for rewrites.  0.000  1.000  0.000   

    compound  
0     0.0000  
1     0.7964  
2    -0.5994  
3    -0.5423  
4     0.0000  
..       ...  
92    0.0000  
93    0.0000  
94    0.3818  
95    0.0000  
96    0.0000  

[97 rows x 5 columns]

In [30]:
# Calcular los totales de 'neg', 'neu' y 'pos' para todas las oraciones
totales_sentimiento = df_sentimiento_oraciones[['neg', 'neu', 'pos']].sum()

# Crear un nuevo DataFrame con los totales
df_totales_sentimiento = pd.DataFrame({'Totales': totales_sentimiento})

# Imprimir el nuevo DataFrame con los totales
df_totales_sentimiento

Totales
neg    3.960
neu   88.497
pos    4.541

In [31]:
# Calcular el total general de 'neg', 'neu' y 'pos'
total_general = totales_sentimiento.sum()

# Calcular los porcentajes respecto al total
df_totales_sentimiento['Porcentaje'] = (totales_sentimiento / total_general * 100).round(1).astype(str) + '%'

# Imprimir el nuevo DataFrame con los totales y porcentajes
df_totales_sentimiento

Totales Porcentaje
neg    3.960       4.1%
neu   88.497      91.2%
pos    4.541       4.7%

### COCLUSIÓN AL ANÁLISIS DE SENTIMIENTO POR ORACIONES.

Según nos muestra la tabla anterior, el análisis de sentimientos por frases revela que: 
- El 91,2% de las oraciones arrojan un sentimiento neutro.
- El 4,1% de las oraciones arrojan un sentimiento negativo.
- 3l 4,7% de las oraciones arrojan un sentimiento positivo.


No hay gran diferencia a los resultados obtenidos en el análisis por palabras. Cabe señalar que el análisis por oraciones asigna algo menos al conjunto de neutralidad, y valora en mayor cantidad el número de construcciones tanto positivas como negativas.

# <font color='blue'> MI CONLUSIÓN PERSONAL
    
    

<font color='blue'>  El texto seleccionado es un texto académico, complejo, y que apunta a un tema controvertido como es la época colonial y las actuales actitudes poscoloniales de las potencias europeas. 
Es un texto plagado de acusaciones y reivindicaciones realiadas de forma sútil amaparado por la dialéctiva académica.
Vistos los resultados que nos arroja el test el análisis de sentimiento para este texto, tengo que decir que el text no ha sido capaz de detectar las acusaciones, ironía e inquina que se esconde tras las frases y palabras del texto.